### PR TITLE
[14.0] Add intercompany shared contact

### PR DESCRIPTION
**Pull request by sebastienbeau**
This module automatically shares all the contacts originating from companies (res.company object).

For example, suppose we have a multi-company environment with two companies (res.company) A and B in Odoo.
We have users uA, uB (belonging to their respective companies).

Normally, uA cannot access the contact B.
With this module, uA can see B (and inversely uB can see A).
uA still can't see all the other (supplier/reseller) contacts of company B.

### CODE REVIEW

File: intercompany_shared_contact/models/res_partner.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                record.company_id = False
+            else:
+                record.origin_company_id = False
+
+    # super().sudo().create() has some interaction that loops infinitely,
+    # thus the switch with multiple sudo() calls
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        if self.env.context.get("creating_res_company"):
```
Comment (sebastienbeau) [inline intercompany_shared_contact/models/res_partner.py]: @kevinkhao This is a huge security issue.
Every time you use sudo becarefull.
With this code anybody without behind logged can create / update a partner.
So basically you can update the email of admin user, use forget password and you are the admin of the database !

Be really carefull !

Comment (sebastienbeau) [reply intercompany_shared_contact/models/res_partner.py]: I am working on a fix

**Comment by sebastienbeau** [comment]
@kevinkhao @bguillot @PierrickBrun can you review ?

**Review by kevinkhao — APPROVED** [review]

**Review by bguillot — APPROVED** [review]
Used in production

**Pull request merged**